test/unit/v1.0.0/converter-v1-to-v21.test.js

```diff
@@ -242,6 +248,200 @@ describe('v1.0.0 to v2.1.0', function () {
+                        expect(JSON.parse(JSON.stringify(converted))).to.eql({
+                            name: '',
+                            request: {
+                                auth: null,
+                                body: { mode: 'raw', raw: '' },
+                                header: []
+                            },
+                            response: []
+                        });
+                        done();
+                    });
+                });
+
+                it('should correctly infer a noauth type from `currentHelper`', function (done) {
+                    var source = {
+                        currentHelper: null,
+                        helperAttributes: { id: 'normal', foo: 'bar' }
+                    };
+
+                    transformer.convertSingle(source, options, function (err, converted) {
+                        expect(err).to.not.be.ok;
+
+                        // remove `undefined` properties for testing
+                        expect(JSON.parse(JSON.stringify(converted))).to.eql({
+                            name: '',
+                            request: {
+                                auth: null,
+                                body: { mode: 'raw', raw: '' },
+                                header: []
+                            },
+                            response: []
+                        });
+                        done();
+                    });
+                });
+
+                it('should correctly infer a noauth type from `currentHelper`, even if auth exists', function (done) {
+                    var source = {
+                        currentHelper: null,
+                        helperAttributes: { id: 'normal', foo: 'bar' },
+                        auth: {
+                            type: 'basic',
+                            basic: { username: 'postman', password: 'password' }
+                        }
+                    };
+
+                    transformer.convertSingle(source, options, function (err, converted) {
+                        expect(err).to.not.be.ok;
+
+                        // remove `undefined` properties for testing
+                        expect(JSON.parse(JSON.stringify(converted))).to.eql({
+                            name: '',
+                            request: {
+                                auth: null,
+                                body: { mode: 'raw', raw: '' },
+                                header: []
+                            },
+                            response: []
+                        });
+                        done();
+                    });
+                });
+            });
+        });
+
+        describe('collections', function () {
+            it('should correctly infer a noauth type from a regular auth object', function (done) {
+                var source = {
+                    auth: { type: 'noauth' },
```

Comment:
@kunagpal even though collections will never have auth type set to `noauth` it is an allowed value. Wheather normalise should remove it is a different question.

Reply:
Empty auth states are not needed for collections, this is being done to reduce the payload size.